### PR TITLE
Diversify the test matrix

### PR DESCRIPTION
**Pull request by pakrym**
OS-TFM combinations were duplicated before.

image

### CODE REVIEW

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 1]
```diff
@@ -5,24 +5,39 @@
...
+        "TestTargetFramework": "net6.0"
       },
-      "macOS-10.15": {
-          "OSVmImage": "macOS-10.15",
-          "Pool": "Azure Pipelines"
+
+      "Windows2019_NET6.0": {
+        "OSVmImage": "MMS2019",
+        "Pool": "azsdk-pool-mms-win-2019-general",
+        "TestTargetFramework": "net6.0"
```
Comment (christothes) [inline eng/pipelines/templates/stages/platform-matrix.json:22]: Is netcoreapp3.1 not included here because it is the default below?

Comment (pakrym) [reply eng/pipelines/templates/stages/platform-matrix.json:22]: Yep, we cover all 3 frameworks on windows this way.

**Comment by weshaggard** [comment]
Is this a result of your Debug/Release configuration change?

**Comment by pakrym** [comment]
>Is this a result of your Debug/Release configuration change?

No, Debug/Release configuration change didn't affect OS-TFM combinations.

File: eng/pipelines/templates/stages/platform-matrix.json (hunk 1 above)
Comment (annelo-msft) [reply eng/pipelines/templates/stages/platform-matrix.json:22]: Where is the entry with the default?

Comment (pakrym) [reply eng/pipelines/templates/stages/platform-matrix.json:22]: https://github.com/Azure/azure-sdk-for-net/pull/26512/files#diff-6d57a4119ae22b0fede4d3aab5dff565cee910416d4eaa647d83e1d749c9b559R50

**Comment by weshaggard** [comment]
Oh it looks like it is because of project vs package references. I don't like the fact that we have to explicitly list all the combinations as that is because it limits our sparse strategy across the matrix. @benbp are there other options? Could we maybe change the order of the parameters to mix this up a little better?

**Comment by pakrym** [comment]
I chatted with @benbp and we came to a conclusion that this is the simplest option to get the coverage we need.

**Review by weshaggard — COMMENTED** [review]
While I don't like the fact, we manually created this sparse matrix I will defer to @benbp to sign-off on this as the best approach.

**Review by benbp — APPROVED** [review]

**Comment by benbp** [comment]
> While I don't like the fact, we manually created this sparse matrix I will defer to @benbp Ben Broderick Phillips FTE to sign-off on this as the best approach.

@weshaggard this is another instance of the problem: "how do we traverse the matrix in a way that does not wrap (i.e. not diagonally)?" I remember last time we explored this problem in depth, it was not as simple as it seemed, because we can't generalize knowing which matrix dimensions to iterate vs. which not to, and exposing those knobs to the user is complicated at best while remaining deterministic.

It might be worth taking another look at it though and trying to come up with some form of solution, perhaps exploring more functionality around sub-matrices (walk a sparse matrix for parameters A and B, but walk through a subset of a full matrix for parameters C and D).

**Pull request merged**